autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingHandler.java

```diff
@@ -1,105 +1,38 @@
 
-import com.sequenceiq.cloudbreak.client.CloudbreakClient;
-import com.sequenceiq.periscope.aspects.AmbariRequestLogging;
 import com.sequenceiq.periscope.domain.BaseAlert;
 import com.sequenceiq.periscope.domain.Cluster;
-import com.sequenceiq.periscope.domain.ScalingPolicy;
 import com.sequenceiq.periscope.log.MDCBuilder;
+import com.sequenceiq.periscope.monitor.ScalingHandlerUtil;
 import com.sequenceiq.periscope.monitor.event.ScalingEvent;
-import com.sequenceiq.periscope.monitor.executor.LoggedExecutorService;
 import com.sequenceiq.periscope.service.ClusterService;
-import com.sequenceiq.periscope.service.RejectedThreadService;
-import com.sequenceiq.periscope.utils.ClusterUtils;
-import com.sequenceiq.periscope.utils.TimeUtil;
 
 @Component
 public class ScalingHandler implements ApplicationListener<ScalingEvent> {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ScalingHandler.class);
-
-    @Inject
-    private LoggedExecutorService loggedExecutorService;
-
     @Inject
     private ClusterService clusterService;
 
     @Inject
-    private ApplicationContext applicationContext;
-
-    @Inject
-    private RejectedThreadService rejectedThreadService;
-
-    @Inject
-    private AmbariRequestLogging ambariRequestLogging;
-
-    @Inject
-    private CloudbreakClient cloudbreakClient;
+    private ScalingHandlerUtil scalingHandlerUtil;
 
     @Override
     public void onApplicationEvent(ScalingEvent event) {
-        BaseAlert alert = event.getAlert();
-        Cluster cluster = clusterService.findById(alert.getCluster().getId());
+        List<BaseAlert> alerts = event.getAlerts();
+        Long clusterId = alerts.stream().findFirst().map(ma -> ma.getCluster().getId()).orElseThrow();
+        Cluster cluster = clusterService.findById(clusterId);
         MDCBuilder.buildMdcContext(cluster);
-        scale(cluster, alert.getScalingPolicy());
-    }
-
-    private void scale(Cluster cluster, ScalingPolicy policy) {
-        long remainingTime = getRemainingCooldownTime(cluster);
-        if (remainingTime <= 0) {
-            int totalNodes = Math.toIntExact(cloudbreakClient.autoscaleEndpoint().getHostMetadataCountForAutoscale(cluster.getStackId(), policy.getHostGroup()));
-            int desiredNodeCount = getDesiredNodeCount(cluster, policy, totalNodes);
-            if (totalNodes != desiredNodeCount) {
-                Runnable scalingRequest = (Runnable) applicationContext.getBean("ScalingRequest", cluster, policy, totalNodes, desiredNodeCount);
-                loggedExecutorService.submit("ScalingHandler", scalingRequest);
-                rejectedThreadService.remove(cluster.getId());
-                cluster.setLastScalingActivityCurrent();
-                clusterService.save(cluster);
-            } else {
-                LOGGER.info("No scaling activity required");
+        alerts.forEach(alert -> {
+            if (scalingHandlerUtil.isCooldownElapsed(cluster)) {
```

Comment:
Are you sure this won't cause any race conditions? I mean two or more possible subsequent "true" answers for isCooldownElapsed..